NetKAN.schema

```diff
@@ -42,39 +42,39 @@
             "type": "string",
             "oneOf": [
                 {
-                    "pattern": "^#/ckan/spacedock/[0-9]+",
+                    "pattern": "^#\/ckan\/spacedock\/[0-9]+$",
                     "description": "Indicates that data should be fetched from SpaceDock"
                 },
                 {
-                    "pattern": "^#/ckan/curse/[A-Za-z0-9_-]+",
+                    "pattern": "^#\/ckan\/curse\/[A-Za-z0-9_-]+$",
                     "description": "Deprecated format, no longer supported"
                 },
                 {
-                    "pattern": "^#/ckan/github/[A-Za-z0-9_-]+/[A-Za-z0-9_-]+(/asset_match/.+|/version_from_asset/.+)?",
+                    "pattern": "^#\/ckan\/github\/[A-Za-z0-9_-]+\/[A-Za-z0-9_-]+(\/asset_match\/.+|\/version_from_asset\/.+)$",
```

Comment:
```suggestion
                    "pattern": "^#/ckan/github/[A-Za-z0-9_-]+/[A-Za-z0-9_-]+(/asset_match/.+|/version_from_asset/.+)?$",
```

Removing the `?` makes the capture group **required**, which is wrong; it's perfectly allowed to leave out both `asset_match` and `version_from_asset`. E.g., this should be considered valid:

```json
{
  "identifier": "ReclaimKspDres",
  "$kref": "#/ckan/github/reclaim-ksp/reclaim-ksp-monorepo"
}
```

Luckily I don't think `?$` will confuse the parser.

Reply:
Ah, entirely true.
And, turns out, having `$` close off the regex is enough for the actual fix I opened this PR for: with `$` present, my example case above still works and the warning squiggle in VSCode is there.

So yes this is the correct change!